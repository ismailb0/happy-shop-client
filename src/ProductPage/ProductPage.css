.product-page-wrapper {
  align-content: center;
  display: flex;
  height: 100%;
  margin: 5% 5%;
  overflow-x: auto;
  padding-top: 5vh;
  padding-bottom: 5vh;
  padding-left: 10vh;
  padding-right: 15vh;
  width: 100%;
}

.product-info-wrapper {
  height: 70vh;
  width: 50vw;
}

.card-page-content {
  font-size: smaller;
  height: 14vh;
  padding: 1vh 1vw;
  text-align: left;
  width: 35vw;
}

.back-button {
  margin-right: 10vw;
}
