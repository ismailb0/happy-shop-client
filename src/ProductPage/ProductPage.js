import React from 'react';
import { Link } from "react-router-dom";
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia'
import ArrowBack from '@material-ui/icons/ArrowBack';

import './ProductPage.css';

const styles = theme => ({
  card: {
    maxWidth: 600,
    maxHeight: 600,
  },
  media: {
    height: "auto",
    maxHeight: "400px",
    width: "auto",
    maxWidth: "600px",
    display: "block",
    marginLeft: "auto",
    marginRight: "auto",
  }
});

function ProductPage({ location, classes }) {

  const { state: { product } } = location;

  return (
    <div className="product-page-wrapper">
      <div className='back-button'>
        <Link to="/">
          <ArrowBack />
        </Link>
      </div>
      <Card className={classes.card}>
        <CardMedia
          component="img"
          className={classes.media}
          image={product.image}
          title={product.name}
        />
        <div className="card-page-content">
          <div className='product-brand'>
            {product.brand}
          </div>
          <div>
            {product.name}
          </div>
          <div className='product-price'>
            {`$${product.price}`}
          </div>
          <div>
            {product.description}
          </div>
        </div>
      </Card>
    </div>
  );
}

export default withStyles(styles)(ProductPage);
