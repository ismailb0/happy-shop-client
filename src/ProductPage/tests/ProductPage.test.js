import React from "react";
import { shallow } from "enzyme";
import { unwrap } from "@material-ui/core/test-utils";
import ProductPage from "../ProductPage";

const ComponentNaked = unwrap(ProductPage);

describe("<ProductPage />", () => {
  it("with shallow", () => {
    const product = {
      brand: "Sephora Collection",
      description: "A gel lip gloss with plumped-up, light-reflecting shine.\r\n",
      image: "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
      name: "Ultra Shine Lip Gel",
      price: 19.0
    };
    const props = {
      location: {
        state: {
          product
        }
      }
    }
    const wrapper = shallow(<ComponentNaked classes={{}} {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
