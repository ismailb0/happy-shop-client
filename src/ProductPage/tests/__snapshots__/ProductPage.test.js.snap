// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPage /> with shallow 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductPage
    classes={Object {}}
    location={
      Object {
        "state": Object {
          "product": Object {
            "brand": "Sephora Collection",
            "description": "A gel lip gloss with plumped-up, light-reflecting shine.
",
            "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
            "name": "Ultra Shine Lip Gel",
            "price": 19,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="back-button"
        >
          <Link
            replace={false}
            to="/"
          >
            <pure(ArrowBackIcon) />
          </Link>
        </div>,
        <WithStyles(Card)>
          <WithStyles(CardMedia)
            component="img"
            image="https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png"
            title="Ultra Shine Lip Gel"
          />
          <div
            className="card-page-content"
          >
            <div
              className="product-brand"
            >
              Sephora Collection
            </div>
            <div>
              Ultra Shine Lip Gel
            </div>
            <div
              className="product-price"
            >
              $19
            </div>
            <div>
              A gel lip gloss with plumped-up, light-reflecting shine.

            </div>
          </div>
        </WithStyles(Card)>,
      ],
      "className": "product-page-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/"
          >
            <pure(ArrowBackIcon) />
          </Link>,
          "className": "back-button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <pure(ArrowBackIcon) />,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(CardMedia)
              component="img"
              image="https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png"
              title="Ultra Shine Lip Gel"
            />,
            <div
              className="card-page-content"
            >
              <div
                className="product-brand"
              >
                Sephora Collection
              </div>
              <div>
                Ultra Shine Lip Gel
              </div>
              <div
                className="product-price"
              >
                $19
              </div>
              <div>
                A gel lip gloss with plumped-up, light-reflecting shine.

              </div>
            </div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": undefined,
              "component": "img",
              "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
              "title": "Ultra Shine Lip Gel",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="product-brand"
                >
                  Sephora Collection
                </div>,
                <div>
                  Ultra Shine Lip Gel
                </div>,
                <div
                  className="product-price"
                >
                  $19
                </div>,
                <div>
                  A gel lip gloss with plumped-up, light-reflecting shine.

                </div>,
              ],
              "className": "card-page-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sephora Collection",
                  "className": "product-brand",
                },
                "ref": null,
                "rendered": "Sephora Collection",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Ultra Shine Lip Gel",
                },
                "ref": null,
                "rendered": "Ultra Shine Lip Gel",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$19",
                  "className": "product-price",
                },
                "ref": null,
                "rendered": "$19",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A gel lip gloss with plumped-up, light-reflecting shine.
",
                },
                "ref": null,
                "rendered": "A gel lip gloss with plumped-up, light-reflecting shine.
",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="back-button"
          >
            <Link
              replace={false}
              to="/"
            >
              <pure(ArrowBackIcon) />
            </Link>
          </div>,
          <WithStyles(Card)>
            <WithStyles(CardMedia)
              component="img"
              image="https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png"
              title="Ultra Shine Lip Gel"
            />
            <div
              className="card-page-content"
            >
              <div
                className="product-brand"
              >
                Sephora Collection
              </div>
              <div>
                Ultra Shine Lip Gel
              </div>
              <div
                className="product-price"
              >
                $19
              </div>
              <div>
                A gel lip gloss with plumped-up, light-reflecting shine.

              </div>
            </div>
          </WithStyles(Card)>,
        ],
        "className": "product-page-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              <pure(ArrowBackIcon) />
            </Link>,
            "className": "back-button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <pure(ArrowBackIcon) />,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(CardMedia)
                component="img"
                image="https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png"
                title="Ultra Shine Lip Gel"
              />,
              <div
                className="card-page-content"
              >
                <div
                  className="product-brand"
                >
                  Sephora Collection
                </div>
                <div>
                  Ultra Shine Lip Gel
                </div>
                <div
                  className="product-price"
                >
                  $19
                </div>
                <div>
                  A gel lip gloss with plumped-up, light-reflecting shine.

                </div>
              </div>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": undefined,
                "component": "img",
                "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
                "title": "Ultra Shine Lip Gel",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="product-brand"
                  >
                    Sephora Collection
                  </div>,
                  <div>
                    Ultra Shine Lip Gel
                  </div>,
                  <div
                    className="product-price"
                  >
                    $19
                  </div>,
                  <div>
                    A gel lip gloss with plumped-up, light-reflecting shine.

                  </div>,
                ],
                "className": "card-page-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sephora Collection",
                    "className": "product-brand",
                  },
                  "ref": null,
                  "rendered": "Sephora Collection",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Ultra Shine Lip Gel",
                  },
                  "ref": null,
                  "rendered": "Ultra Shine Lip Gel",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "$19",
                    "className": "product-price",
                  },
                  "ref": null,
                  "rendered": "$19",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A gel lip gloss with plumped-up, light-reflecting shine.
",
                  },
                  "ref": null,
                  "rendered": "A gel lip gloss with plumped-up, light-reflecting shine.
",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
